### ISSUE TITLE
Invalid js file when compiling with AUDIO_WORKLET + MODULARIZE

### ISSUE THREAD

**Issue by mineevgleb** [emscripten-core/emscripten#19796]
Version of emscripten: 3.1.42

Hello! 

I'm trying to build a small wasm module to get familiar with AudioWorklets. The code itself is pretty minimal, I made a class that does the initialisation, and starts the playback, and then exposed it with embind. It is basically a slight modification of the tutorial here https://emscripten.org/docs/api_reference/wasm_audio_worklets.html

I use following linker flags to enable AudioWorklets and export an ES6 module:

```
target_link_options(wasm_audio_worklet_test PUBLIC
    "-sAUDIO_WORKLET=1"
    "-sWASM_WORKERS=1"
    "-sEXPORT_ES6=1"
    "-sMODULARIZE=1")
```

When I try to import a resulting wasm_audio_worklet_test.js file I'm getting an error: 
SyntaxError: Unexpected identifier 'globalThis' (at wasm_audio_worklet_test.js:3989:1)

The structure of the generated file looks like this:

```
var Module = (() => {
  var _scriptDir = import.meta.url;
  
  return (
async function(moduleArg = {}) {

... lots of code ...

  return moduleArg.ready
}
globalThis.AudioWorkletModule = Module; // <- Syntax error here
);
})();
export default Module;
```

So, the error is due to the statement `  return (async function(moduleArg = {}) { ... } globalThis.AudioWorkletModule = Module;);`

And looking at https://github.com/emscripten-core/emscripten/blob/main/emcc.py:3909 this is how it always will be generated if AUDIO_WORKLET and MODULARIZE both specified.

Am I missing something or is this a bug?

**Comment by mineevgleb** [comment]
> I had to move the line with globalThis before the previous } to avoid runtime error.

Yeah, I also tried that. But it still didn't work for me. When execution comes to audio worklet (BootstrapMessages class), here
```
if (globalThis.AudioWorkletModule) {
    AudioWorkletModule(Module); 
    delete globalThis.AudioWorkletModule; 
}
```
globalThis.AudioWorkletModule is not defined (Presumably because it's a different thread? Don't know too much about web workers)

**Comment by stefanholmer** [comment]
I also ran into this issue. Unsure if it's a user error or a bug?

**Comment by sbc100** [comment]
IIRC there is no `globalThis` in ES6 modules?  Is that right?

**Comment by juj** [comment]
Ah right, it is probably the `-sEXPORT_ES6=1` that is causing the issue. I don't recall if I tested that flag combination, and reading the test code, looks like that combination is not tested in the test suite either. We should add a new test for that flag into audio worklets.

Does it work (around the issue) if you drop `-sEXPORT_ES6=1`?

**Comment by stefanholmer** [comment]
From my experience this issue occurs as soon as you build with -sMODULARIZE=1 as I didn't have -sEXPORT_ES6=1 in my build config.

**Comment by imiskolee** [comment]
I has same issue.

**Comment by gregor-poloczek** [comment]
Same here. I used these minimal settings:
```
em++ \
  -s AUDIO_WORKLET=1 \
  -s WASM_WORKERS=1 \
  -s MODULARIZE=1 \
  -s 'EXPORT_NAME="experimentAudio"' \
  ./src/main.cpp \
  -o output/experiment-audio.js
```

**Comment by sbc100** [comment]
I believe the fix is already in flight: #20249
